plugins/BEdita/Core/src/Shell/StreamsShell.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2019 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace BEdita\Core\Shell;
+
+use BEdita\Core\Model\Entity\Stream;
+use Cake\Console\Shell;
+use Cake\I18n\Date;
+use Cake\ORM\TableRegistry;
+
+/**
+ * Stream shell commands: removeOrphans
+ *
+ * @since 4.0.0
+ */
+class StreamsShell extends Shell
+{
+
+    /**
+     * {@inheritDoc}
+     *
+     * @codeCoverageIgnore
+     */
+    public function getOptionParser()
+    {
+        $parser = parent::getOptionParser();
+        $parser->addSubcommand('removeOrphans', [
+            'help' => 'remove obsolete/orphans streams and related files',
+            'parser' => [
+                'description' => [
+                    'Remove orphans streams.'
+                ],
+                'options' => [
+                    'days' => [
+                        'help' => 'Days to consider for stream research for orphans',
+                        'required' => false,
+                        'default' => 1,
+                    ],
+                ],
+            ]
+        ]);
+
+        return $parser;
+    }
+
+    /**
+     * Create a new Stream
+     *
+     * @return void
+     */
+    public function removeOrphans()
+    {
+        $days = (int)$this->param('days');
+        $streamsTable = TableRegistry::get('Streams');
+        $query = $streamsTable
+            ->find()
+            ->where([
+                'object_id IS NULL',
+                'created >=' => new \DateTime(sprintf('-%d days', $days)),
```

Comment:
I think it would make much more sense if the comparison was made the other way around: delete streams **older** than _N_ days:
* you wouldn't need to pass a very big integer to dispose of very old streams
* you wouldn't break things for a client that's uploading a stream right now, that would see its stream deleted for no reason

Reply:
Also: I'd prefer to use `Cake\I18n\Time::now()->subDays($days)`, as it's easier to test thanks to `Time::setTestNow()`. Otherwise, testing would be quite a pain.